### ISSUE TITLE
Instant loading: returning to index page breaks search and other features

### ISSUE THREAD

**Issue by sarcastro** [squidfunk/mkdocs-material#6275]
### Bug description

Search is not functional from the index page when `navigation.instant` is enabled except for when first loaded.

Refreshing page restores functionality.

_Note: issue does not appear to be limited to search as page no longer refreshes automatically when using `mkdocs serve` either._

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Creating a reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/)
- [Setting up navigation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/)
- [Setting up navigation: Instant loading](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-loading)
  - _"Now, the search index survives navigation..."_ – issue appears to contradict this

### Reproduction

[9.4.7-navigation-instant-index.zip](https://github.com/squidfunk/mkdocs-material/files/13231550/9.4.7-navigation-instant-index.zip)

### Steps to reproduce

1. Navigate away from index page (e.g. by following a link)
2. Return to index page (e.g. using left navigation menu)
3. Try to search (no longer functional)

_Note: steps to reproduce are included in the reproduction as well._

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! I think this should be fixed with 494cae1e3.

**Comment by sarcastro** [comment]
> Thanks for reporting! I think this should be fixed with [494cae1](https://github.com/squidfunk/mkdocs-material/commit/494cae1e36664a5d106c1371b05e74f90703b919)!

Fantastic, much appreciated!

**Comment by squidfunk** [comment]
Released as part of [9.4.8](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.8).

**Comment by sarcastro** [comment]
> Released as part of [9.4.8](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.8).

Didn't seem to make a difference, unfortunately. Here's a new reproduction with the same configuration and steps to reproduce as before.

[9.4.8-navigation-instant-index-2.zip](https://github.com/squidfunk/mkdocs-material/files/13285301/9.4.8-navigation-instant-index-2.zip)

**Comment by sarcastro** [comment]
@squidfunk - should I create a duplicate report?

**Comment by squidfunk** [comment]
No, I'll look into it once I find some time.

**Comment by squidfunk** [comment]
Fixed in https://github.com/squidfunk/mkdocs-material-insiders/pull/79 – the fixes are currently only in Insiders, so anybody having access can already test them. I still need to do some minor refactoring that shouldn't impact the functionality, and after that, I'll backport the fixes to the community edition. I just wanted to give them into the hands of users as soon as possible. This should resolve all known problems with instant loading, at least I've tested this and all other issues.
